File: 26. Reverse words in a given string - GFG/26.-reverse-words-in-a-given-string.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends


class Solution
{
    public:
    //Function to reverse words in a given string.
    string reverseWords(string S) { 
        // code here 
        string s = "",temp = "";
        
        for (int i = (int)S.length()-1; i>-1; i--){
            if(S[i]!= '.') temp+= S[i];
            
            else{
                reverse(temp.begin(),temp.end());
                s += temp;
                s += '.';
                temp = "";
            }
        }
        reverse(temp.begin(),temp.end());
        s += temp;
        return s;
        
    } 
};

// { Driver Code Starts.
int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        Solution obj;
        cout<<obj.reverseWords(s)<<endl;
    }
}  // } Driver Code Ends

File: 26. Reverse words in a given string - GFG/26.-reverse-words-in-a-given-string.py

# User function Template for python3

class Solution:
    
    #Function to reverse words in a given string.
    def reverseWords(self,S):
        return ".".join(S.split('.')[::-1])
        #Python 1 liner solution

#{ 
#  Driver Code Starts
# Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        s = str(input())
        obj = Solution()
        print(obj.reverseWords(s))

# } Driver Code Ends